### PR TITLE
Plotting improvements and further evaluation improvements

### PR DESCRIPTION
**Pull request by NiklasKoehnecke**

### CODE REVIEW

File: wikipedia_cleanup/evaluation.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+from typing import Any, List, Tuple
+
+import numpy as np
+import pandas as pd
+from matplotlib import pyplot as plt
+from sklearn.metrics import precision_recall_fscore_support
+
+
+def stats_to_string(
+    pre_rec_f1_stat: Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray],
```
Comment (MartinBuessemeyer) [inline wikipedia_cleanup/evaluation.py]: Isn't that precision, recall, f1 and support? Then maybe add _supp to the name.

File: wikipedia_cleanup/evaluation.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                zero_division=0,
+            )
+            for i in range(current_labels.shape[1])
+        ]
+        prec = np.array([stat[0][0] for stat in stats])
+        rec = np.array([stat[1][0] for stat in stats])
+        plt.figure()
+        if timeframe == 1:
+            average = 5
+            prec = np.mean(np.reshape(prec, (-1, average)), axis=1)
```
Comment (MartinBuessemeyer) [inline wikipedia_cleanup/evaluation.py:197]: You assume here that the shape is dividable by 5. That might currently be the case, but might fail later on when choosing a different test set size. Maybe add padding?

File: wikipedia_cleanup/predict.py [hunk 3]
```diff
@@ -36,13 +42,19 @@ def __init__(
...
+        own_relevant_attributes = ["value_valid_from", "template"]
         self.relevant_attributes = list(
             set(own_relevant_attributes)
             | set(predictor.get_relevant_attributes())
             | set(self.group_key)
         )
         self.data: pd.DataFrame = pd.DataFrame()
+        self.run_results: Dict[str, Any] = {}
+        current_time = datetime.now()
+        self.run_id = (
```
Comment (MartinBuessemeyer) [inline wikipedia_cleanup/predict.py:51]: I would rather add an optional argument to the constructor and take the time format as a backup.

File: wikipedia_cleanup/evaluation.py (hunk 2 above)
Comment (NiklasKoehnecke) [reply wikipedia_cleanup/evaluation.py:197]: Solved, i check if its divisible by 5, otherwise i skip it
